### ISSUE TITLE
tf.keras.optimizers.experimental.AdamW only support constant weight_decay

### ISSUE THREAD

**Issue by x10000year** [keras-team/tf-keras#614]
tf.keras.optimizers.experimental.AdamW only support constant weight_decay. The legacy tfa.optimizers.AdamW supports callable weight_decay, which is much better.

```
model = MyModel()
schedule = tf.keras.optimizers.schedules.PolynomialDecay(
    1.0,
    decay_steps=10000)
learning_rate_fn = lambda: learning_rate * schedule(model.iterations.read_value())
weight_decay_fn = lambda: weight_decay * schedule(model.iterations.read_value())
optimizer = tfa.optimizers.AdamW(weight_decay=weight_decay_fn,
                                 learning_rate=learning_rate_fn)
```

As above code shows, tfa.optimizers.AdamW allows us to specify the schedule of weight decay, which should be in proportional with learning rate schedule. However, I cannot do the same thing with tf.keras.optimizers.experimental.AdamW.
